Add tracking to homepage

chao-xian opened the pull request:
, [Jira issue NAV-2910](https://gov-uk.atlassian.net/browse/NAV-2910)⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What/Why

Adding tracking to the homepage to get baseline data prior to an update to the design.

https://trello.com/c/OuT7fPEL

(Note: Each commit corresponds to a row in the spreadsheet attached from the Trello card)

## Visual changes
none

chao-xian committed:
Add tracking to search box on homepage

Adding tracking to the homepage to get baseline data prior to an update to the design.

chao-xian committed:
Add tracking to the Popular links on homepage

Adding tracking to the homepage to get baseline data prior to an update to the design.

chao-xian committed:
Add tracking to mainstream browse topics on Homepage

Adding tracking to the homepage to get baseline data prior to an update to the design.

chao-xian committed:
Add tracking to the Big Numbers on Homepage

Adding tracking to the homepage to get baseline data prior to an update to the design.

chao-xian committed:
Add tracking to Whitehall links on Homepage

Adding tracking to the homepage to get baseline data prior to an update to the design.

chao-xian committed:
Add tracking to promo images

Adding tracking to the homepage to get baseline data prior to an update to the design.

chao-xian committed:
Add tracking to promo text links

Adding tracking to the homepage to get baseline data prior to an update to the design.

chao-xian committed:
Add tracking to Most active links

Adding tracking to the homepage to get baseline data prior to an update to the design.

owenatgov commented on app/views/homepage/index.html.erb:
```diff
@@ -21,21 +21,83 @@
+            <%
+              link_class = "govuk-link govuk-link--inverse home-top__link-item"
+              items = []
+              items[0] = link_to(t("homepage.index.covid"),
```
Personal prefs thing, `items << link_to...` instead of `items[n] = link_to...` would anonymise the array keys in this separated declaration. I wonder if this would reduce the tiny little bit of extra cognisance of adding an extra number to the items array as well as writing a new list item if someone did need to add stuff to the list.

chao-xian replied:
I did think about that but I wanted to ensure order of the links. I know I could order the pushes in the code that way... 🤔

chao-xian replied:
Actually you're right it'd be easier to maintain. I thought being explicit would make it clearer about the order. Have amended.

owenatgov commented on app/views/homepage/index.html.erb:
```diff
@@ -57,17 +119,29 @@
+                 data-track-category="homepageClicked"
+                 data-track-action="departmentsLink"
+                 data-track-label="/government/organisations#ministerial_departments"
+                 data-track-dimension="<%= t('homepage.index.ministerial_departments_count') %> <%= t('homepage.index.ministerial_departments') %>"
```
Oof. I've been picking away at a big number component in my spare time. Good catch here that I should add tracking options before I merge it and start deploying it across apps 😂

owenatgov reviewed (approved):
Good to go. One little pref comment but nothing major.

chao-xian commented:
Just waiting for Performance Analyst to check it's ok before merging.

chao-xian committed:
Add tracking to Bank Holiday fake-promo

Adding tracking to the homepage to get baseline data prior to an update to the design.

chao-xian committed:
Add tracking to Bank Holiday fake promo text link

Adding tracking to the homepage to get baseline data prior to an update to the design.
